conans/model/build_info_components.py

```diff
@@ -0,0 +1,203 @@
+
+from conans.errors import ConanException
+
+DEFAULT_INCLUDE = "include"
+DEFAULT_LIB = "lib"
+DEFAULT_BIN = "bin"
+DEFAULT_RES = "res"
+DEFAULT_SHARE = "share"
+DEFAULT_BUILD = ""
+
+
+class Component(object):
+
+    def __init__(self, name, root_folder):
+        self._rootpath = root_folder
+        self._name = name
+        self.deps = []
+        self._lib = None
+        self._exe = None
+        self.system_deps = []
+        self.includedirs = [DEFAULT_INCLUDE]
+        self.libdirs = [DEFAULT_LIB]
+        self.resdirs = [DEFAULT_RES]
+        self.bindirs = [DEFAULT_BIN]
+        self.builddirs = [DEFAULT_BUILD]
+        self.srcdirs = []
+        self.defines = []
+        self.cflags = []
+        self.cxxflags = []
+        self.sharedlinkflags = []
+        self.exelinkflags = []
+        self.filter_empty = True
+
+    def as_dict(self):
+        result = {}
+        for name in ["name", "rootpath", "deps", "lib", "exe", "system_deps",
+                     "includedirs", "srcdirs", "libdirs", "bindirs", "builddirs", "resdirs",
+                     "defines", "cflags", "cxxflags", "sharedlinkflags", "exelinkflags"]:
+            result[name] = getattr(self, name)
+        return result
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def rootpath(self):
+        return self._rootpath
+
+    @property
+    def lib(self):
+        return self._lib
+
+    @lib.setter
+    def lib(self, name):
+        assert isinstance(name, str), "'lib' attribute should be a string"
+        if self._exe:
+            raise ConanException("'.exe' is already set for this Component")
+        self._lib = name
+
+    @property
+    def exe(self):
+        return self._exe
+
+    @exe.setter
+    def exe(self, name):
+        assert isinstance(name, str), "'exe' attribute should be a string"
+        if self._lib:
+            raise ConanException("'.lib' is already set for this Component")
+        self._exe = name
+
+
+class DepComponent(object):
+
+    def __init__(self, component):
+        self._rootpath = component.rootpath
+        self._name = component.name
+        self._deps = component.deps
+        self._lib = component.lib
+        self._exe = component.exe
+        self._system_deps = component.system_deps
+        self._includedirs = component.includedirs
+        self._libdirs = component.libdirs
+        self._resdirs = component.resdirs
+        self._bindirs = component.bindirs
+        self._builddirs = component.builddirs
+        self._srcdirs = component.srcdirs
+        self._defines = component.defines
+        self._cflags = component.cflags
+        self._cxxflags = component.cxxflags
+        self._sharedlinkflags = component.sharedlinkflags
+        self._exelinkflags = component.exelinkflags
+        self._filter_empty = component.filter_empty
+
+    def _filter_paths(self, paths):
+        abs_paths = [os.path.join(self._rootpath, p) for p in paths]
```

Comment:
should it have check for `is_relative`? what if path is absolute thing like `/usr/include/something`?

Reply:
From: https://docs.python.org/2/library/os.path.html
> **os.path.join(path, paths)**
> Join one or more path components intelligently. The return value is the concatenation of path and any members of paths with exactly one directory separator (os.sep) following each non-empty part except the last, meaning that the result will only end in a separator if the last part is empty. **If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.**

Reply:
great! thanks for the info, didn't know that